### PR TITLE
[TACHYON-648] Add unit test of EvictorFactory

### PR DESCRIPTION
**Pull request by luoli523**
[TACHYON-648] Add unit test of EvictorFactory

Change the code to meet comments in 
https://github.com/amplab/tachyon/pull/1119

### CODE REVIEW

**Comment by apc999** [comment]
+1 LGTM.

PS, any particular reason to open new PR and move new commits here rather than appending commits to #1119 ?

**Comment by luoli523** [comment]
sorry, I will append commits to the old PR next time .

**Comment by apc999** [comment]
No worry.  Thanks for the patch!

File: servers/src/test/java/tachyon/worker/block/evictor/EvictorFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ */
+public class EvictorFactoryTest {
+  private static BlockMetadataManager sBlockMetadataManager;
+  private static BlockMetadataManagerView sBlockMetadataManagerView;
+  
+  @Rule
+  public TemporaryFolder mTestFolder = new TemporaryFolder();
+
+  @Before
+  public void before() throws Exception {
```
Comment (gpang) [inline servers/src/test/java/tachyon/worker/block/evictor/EvictorFactoryTest.java]: should this be a beforeClass()? (only do this once before all tests?)

File: servers/src/test/java/tachyon/worker/block/evictor/EvictorFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        sBlockMetadataManager = EvictorTestUtils.defaultMetadataManager(
+           tempFolder.getAbsolutePath());
+      }
+      sBlockMetadataManagerView = new BlockMetadataManagerView(sBlockMetadataManager, 
+          Collections.<Integer>emptySet(), Collections.<Long>emptySet());
+    }
+  }
+  
+  @Test
+  public void createGreedyAllocatorTest() {
```
Comment (gpang) [inline servers/src/test/java/tachyon/worker/block/evictor/EvictorFactoryTest.java]: Can you replace "Allocator" with "Evictor"?

File: servers/src/test/java/tachyon/worker/block/evictor/EvictorFactoryTest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    Evictor evictor = EvictorFactory.create(EvictorType.GREEDY, sBlockMetadataManagerView);
+    Assert.assertTrue(evictor instanceof GreedyEvictor);
+  }
+  
+  @Test
+  public void createLRUAllocatorTest() {
```
Comment (gpang) [inline servers/src/test/java/tachyon/worker/block/evictor/EvictorFactoryTest.java]: Can you replace "Allocator" with "Evictor"?

File: servers/src/test/java/tachyon/worker/block/evictor/EvictorFactoryTest.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    Evictor evictor = EvictorFactory.create(EvictorType.LRU, sBlockMetadataManagerView);
+    Assert.assertTrue(evictor instanceof LRUEvictor);
+  }
+  
+  @Test
+  public void createDefaultAllocatorTest() {
```
Comment (gpang) [inline servers/src/test/java/tachyon/worker/block/evictor/EvictorFactoryTest.java]: Can you replace "Allocator" with "Evictor"?

File: servers/src/test/java/tachyon/worker/block/evictor/EvictorFactoryTest.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import tachyon.worker.block.BlockMetadataManager;
+import tachyon.worker.block.BlockMetadataManagerView;
+
+/*
+ * Test EvictorFactory by passing different AllocatorType
```
Comment (gpang) [inline servers/src/test/java/tachyon/worker/block/evictor/EvictorFactoryTest.java]: Can you replace "Allocator" with "Evictor"?

**Comment by gpang** [comment]
Thanks for the patch @luoli523 ! I left some minor comments and questions.

**Comment by apc999** [comment]
This PR stands for too long. Merged it and create  [TACHYON-723 - Fix wrong testcase name in EvictorFactoryTest.java](https://tachyon.atlassian.net/browse/TACHYON-723) to fix this unit tests

**Pull request merged**